libraries/tchaputils/src/main/java/fr/gouv/tchap/libraries/tchaputils/TchapPatterns.kt

```diff
@@ -0,0 +1,225 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2024. DINUM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+ * OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fr.gouv.tchap.libraries.tchaputils
+
+import java.util.Locale
+
+object TchapPatterns {
+
+    /**
+     * Get the homeserver name of a matrix identifier.
+     * The identifier type may be any matrix identifier type: user id, room id, ...
+     * For example in case of "@jean-philippe.martin-modernisation.fr:matrix.test.org", this will return "matrix.test.org".
+     * in case of "!AAAAAAA:matrix.test.org", this will return "matrix.test.org".
+     *
+     * @param mxId the matrix identifier.
+     * @return the homeserver name, if any.
+     */
+    fun getHomeserverNameFromMxId(mxId: String): String = mxId.substringAfter(":", "")
+
+    /**
+     * Get the Tchap display name of the homeserver mentioned in a matrix identifier.
+     * The identifier type may be any matrix identifier type: user id, room id, ...
+     * The returned name is capitalize.
+     * The Tchap HS display name is the component mentioned before the suffix "tchap.gouv.fr"
+     * For example in case of "@jean-philippe.martin-modernisation.fr:name1.tchap.gouv.fr", this will return "Name1".
+     * in case of "@jean-philippe.martin-modernisation.fr:agent.name2.tchap.gouv.fr", this will return "Name2".
+     *
+     * @param mxId the matrix identifier.
+     * @return the Tchap display name of the homeserver.
+     */
+    fun getHomeserverDisplayNameFromMxId(mxId: String): String {
+        var homeserverName = getHomeserverNameFromMxId(mxId)
+        if (homeserverName.contains("tchap.gouv.fr")) {
+            homeserverName.split("\\.".toRegex()).let {
+                if (it.size >= 4) homeserverName = it[it.size - 4]
+            }
+        }
+        return homeserverName.replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }
```

Comment:
Dasn replaceFirstChar, je pense que l'on se retrouve avec le nom complet copié dans le 1er char. 
`finances` -> `Financesinances` ?
Testé dans les Unit Tests ?

Reply:
Dans ton cas avec finances `it = 'f'`
Comme `it` est en minuscule on le remplace par lui même en majuscule.

Pour info on utilisait la méthode `String.capitalize() `en kotlin pour faire cela. Cependant elle est dépréciée en faveur de : `@Deprecated("Use replaceFirstChar instead.", ReplaceWith("replaceFirstChar { if (it.isLowerCase()) it.titlecase(locale) else it.toString() }"))`